rewriter/cleanup.cc

```diff
@@ -0,0 +1,45 @@
+#include "rewriter/cleanup.h"
+#include "ast/Helpers.h"
+#include "ast/ast.h"
+#include "ast/treemap/treemap.h"
+#include "core/core.h"
+
+using namespace std;
+
+namespace sorbet::rewriter {
+
+// This pass gets rid of some unnecessary nodes that are likely to have gotten created in the course of the rewriter
+// pass, specifically by removing EmptyTree nodes whenever they are created
+struct CleanupWalk {
+    unique_ptr<ast::Expression> postTransformInsSeq(core::Context ctx, unique_ptr<ast::InsSeq> insSeq) {
+        ast::InsSeq::STATS_store newStore;
+        for (auto &m : insSeq->stats) {
+            if (!ast::isa_tree<ast::EmptyTree>(m.get())) {
+                newStore.emplace_back(move(m));
+            }
+        }
+        if (newStore.empty()) {
+            return move(insSeq->expr);
+        }
+        insSeq->stats = std::move(newStore);
+        return insSeq;
+    }
+
+    unique_ptr<ast::Expression> postTransformClassDef(core::Context ctx, unique_ptr<ast::ClassDef> classDef) {
+        ast::ClassDef::RHS_store newStore;
+        for (auto &m : classDef->rhs) {
+            if (!ast::isa_tree<ast::EmptyTree>(m.get())) {
+                newStore.emplace_back(move(m));
+            }
+        }
+        classDef->rhs = std::move(newStore);
```

Comment:
Would it be better to use `std::remove_if` and `erase` here instead?

Reply:
Papertrail: we talked about this and I'm erring on the current implementation (as it's both easy to read and ensures that we compact things that have lots of `EmptyTree`s) but it's easy to change in the future.